platform/android/export/gradle_export_util.h

```diff
@@ -242,4 +254,190 @@ String _get_application_tag(const Ref<EditorExportPreset> &p_preset, const Strin
+	String begin_tag = "//ASSET_PACK_INFO_START !!!DO NOT EDIT THIS LINE!!!";
+	String end_tag = "//ASSET_PACK_INFO_END !!!DO NOT EDIT THIS LINE!!!";
+	int b_start = build_gradle_file_string.find(begin_tag);
+	int b_end = build_gradle_file_string.find(end_tag);
+	int b_len = build_gradle_file_string.length();
+	int s_start = settings_gradle_file_string.find(begin_tag);
+	int s_end = settings_gradle_file_string.find(end_tag);
+	int s_len = settings_gradle_file_string.length();
+	print_line(vformat("bstart %d bend %d blen %d\n", b_start, b_end, b_len));
+	print_line(vformat("sstart %d send %d slen %d", s_start, s_end, s_len));
+
+	String b_write = build_gradle_file_string.substr(0, b_start) +
+					 build_gradle_string + build_gradle_file_string.substr(b_end, b_len - b_end);
+	String s_write = settings_gradle_file_string.substr(0, s_start) +
+					 settings_gradle_string + settings_gradle_file_string.substr(s_end, s_len - s_end);
+
+	store_string_at_path("res://android/build/build.gradle", b_write);
+	store_string_at_path("res://android/build/settings.gradle", s_write);
+
+	return OK;
+}
+
+Error _deal_with_pack_files(const String &path, const AssetPackInfo &packInfo) {
+	print_line("path: res://" + path);
+	DirAccess *da = DirAccess::open("res://" + path);
+	if (!da) {
+		return ERR_CANT_OPEN;
+	}
+	da->list_dir_begin();
+	while (true) {
+		String file = da->get_next();
+		print_line(file);
+		if (file == "") {
+			break;
+		}
+		if (file == "." || file == "..") {
+			continue;
+		}
+		if (file.ends_with(".pck")) {
+			String dest = "res://android/build/" + packInfo.packName + "/src/main/assets/" + path + file;
+			da->make_dir_recursive(dest.get_base_dir());
+			print_line("destination: " + dest);
+			da->copy(path + file, dest);
+		} else if (da->dir_exists(file)) {
+			Error err = _deal_with_pack_files(path + file + "/", packInfo);
+			if (err != OK) {
+				da->list_dir_end();
+				return err;
+			}
+		}
+	}
+	da->list_dir_end();
+	return OK;
+}
+
+Error _copy_asset_directory_files(const AssetPackInfo &packInfo) {
+	DirAccess *da = DirAccess::create(DirAccess::ACCESS_RESOURCES);
+	String asset_path_from = "res://android/build/assets/" + packInfo.filepath;
+	String asset_path_to = "res://android/build/" + packInfo.packName + "/src/main/assets/" + packInfo.filepath;
+	print_line(asset_path_to);
+	print_line(asset_path_from);
+	da->make_dir_recursive(asset_path_to);
+	//	Error err = da->copy(asset_path_from, asset_path_to);
+	Error err = da->rename(asset_path_from, asset_path_to);
+	if (err != OK) {
+		print_line("could not rename");
+		return err;
+	}
+	print_line("dealing with pack files now");
+	err = _deal_with_pack_files(packInfo.filepath, packInfo);
+
+	return err;
+}
+
+Error _create_asset_pack_folder(const AssetPackInfo &packInfo) {
+	String packName = packInfo.packName;
+	String deliveryType = "install-time"; //default
+	if (packInfo.deliveryMode == INSTALL_TIME) {
```

Comment:
The next 3 lines don't do anything. An alternative would be a switch or if/else-if/etc where the else (unknown mode) returns an error.